test/cpp/qps/server_async.cc

```diff
@@ -316,8 +316,25 @@ class AsyncQpsServerTest : public Server {
   TestService::AsyncService async_service_;
   std::forward_list<ServerRpcContext *> contexts_;
 
-  std::mutex shutdown_mutex_;
-  bool shutdown_;
+  class PerThreadShutdownState {
+   public:
+    PerThreadShutdownState() : shutdown_(false) {}
+
+    bool shutdown() const {
+      std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
I'm wondering why this should be const - the use in no way requires it, AFAICT, and making it const makes us add mutable, which to the best of my knowledge, we're not yet using anywhere.